Running Dependabot@2 causing 500 internal sql exception via pipeline in ADO

**Describe the bug**
I've added the dependabot extension to our ADO.  When I try to go and run the pipeline to run dependabot, it gives me the following error:

🌎 🠊 [POST] https://dev.azure.com/EnvisionTechnologyPartners/WebIZ/_apis/git/repositories/WebIZ.Graph/pullrequests
🌎 🠈 [500] Internal Server Error
{"$id":"1","innerException":null,"message":"TF400898: An Internal Error Occurred. Activity Id: b7ecf655-43d8-42bc-bf5b-e093cf56dce6.","typeName":"Microsoft.Data.SqlClient.SqlException, Microsoft.Data.SqlClient","typeKey":"SqlException","errorCode":0,"eventId":0}
##[error]Failed to create pull request: Error: Request to 'https://dev.azure.com/EnvisionTechnologyPartners/WebIZ/_apis/git/repositories/WebIZ.Graph/pullrequests' failed: 500 Internal Server Error
Error: Request to 'https://dev.azure.com/EnvisionTechnologyPartners/WebIZ/_apis/git/repositories/WebIZ.Graph/pullrequests' failed: 500 Internal Server Error
    at AzureDevOpsWebApiClient.restApiRequest (/home/vsts/work/_tasks/dependabot_d98b873d-cf18-41eb-8ff5-234f14697896/2.36.1003/utils/azure-devops/AzureDevOpsWebApiClient.js:521:23)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async AzureDevOpsWebApiClient.restApiPost (/home/vsts/work/_tasks/dependabot_d98b873d-cf18-41eb-8ff5-234f14697896/2.36.1003/utils/azure-devops/AzureDevOpsWebApiClient.js:498:16)
    at async AzureDevOpsWebApiClient.createPullRequest (/home/vsts/work/_tasks/dependabot_d98b873d-cf18-41eb-8ff5-234f14697896/2.36.1003/utils/azure-devops/AzureDevOpsWebApiClient.js:188:33)
    at async DependabotOutputProcessor.process (/home/vsts/work/_tasks/dependabot_d98b873d-cf18-41eb-8ff5-234f14697896/2.36.1003/utils/dependabot-cli/DependabotOutputProcessor.js:67:42)
    at async DependabotCli.update (/home/vsts/work/_tasks/dependabot_d98b873d-cf18-41eb-8ff5-234f14697896/2.36.1003/utils/dependabot-cli/DependabotCli.js:113:51)
    at async run (/home/vsts/work/_tasks/dependabot_d98b873d-cf18-41eb-8ff5-234f14697896/2.36.1003/index.js:75:57)

I am able to see that the branches get created successfully however the PRs fail to get created.  I've reached out to MSFT and they've asked me to contact Tingle instead since they believe the error is happening when the request is coming from Dependabot.

**Categorization**
- [ ] This is not a permissions issue (Seek help at https://github.com/tinglesoftware/dependabot-azure-devops/discussions/1245)

**Repository**
[URL: e.g. https://dev.azure.com/tingle/dependabot/_git/repro-684](https://dev.azure.com/EnvisionTechnologyPartners/WebIZ/_git/WebIZ.Graph?path=%2F&version=GBfeatures%2Fmxiong%2F111908_dependabot&_a=contents)

<!-- If credentials are required, send them to support@tingle.software -->
I'm more than happy to do a screen share upon request.

**To Reproduce**
Steps to reproduce the behavior:
1.  Run the pipeline

**Expected behavior**
I expect dependabot to successfully run and create the PRs for review.  At the moment, it is only creating the branches and then failing on a 500 error as described above.

**Logs and screenshots**
[If applicable, include relevant logs or screenshots to help explain your problem.
See [extension troubleshooting](https://github.com/tinglesoftware/dependabot-azure-devops/blob/main/docs/extension.md#troubleshooting-issues) for more on how to collect additional diagnostic logs.
]
https://app.screencast.com/H9yNtrJEknYSh
https://app.screencast.com/acJoiTt6DtxRi




**Extension (please complete the following information):**
- Dependabot@2

**Server (please complete the following information):**
N/A

**Additional context**
Add any other context about the problem here.

--- a/extension/tasks/dependabotV2/utils/azure-devops/AzureDevOpsWebApiClient.test.ts
+++ b/extension/tasks/dependabotV2/utils/azure-devops/AzureDevOpsWebApiClient.test.ts
@@ -1,0 +1,95 @@
+import { jest } from '@jest/globals';
+
+import { VersionControlChangeType } from 'azure-devops-node-api/interfaces/TfvcInterfaces';
+
+import { AzureDevOpsWebApiClient } from './AzureDevOpsWebApiClient';
+import { ICreatePullRequest } from './interfaces/IPullRequest';
+import exp = require('constants');
+
+jest.mock('azure-devops-node-api');
+jest.mock('azure-pipelines-task-lib/task');
+
+describe('AzureDevOpsWebApiClient', () => {
+  const organisationApiUrl = 'https://dev.azure.com/mock-organization';
+  const accessToken = 'mock-access-token';
+  let client: AzureDevOpsWebApiClient;
+
+  beforeEach(() => {
+    client = new AzureDevOpsWebApiClient(organisationApiUrl, accessToken);
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('createPullRequest', () => {
+    let pr: ICreatePullRequest;
+
+    beforeEach(() => {
+      pr = {
+        project: 'project',
+        repository: 'repository',
+        source: {
+          branch: 'update-branch',
+          commit: 'commit-id',
+        },
+        target: {
+          branch: 'main',
+        },
+        title: 'PR Title',
+        description: 'PR Description',
+        commitMessage: 'Commit Message',
+        changes: [
+          {
+            path: 'file.txt',
+            content: 'hello world',
+            encoding: 'utf-8',
+            changeType: VersionControlChangeType.Add,
+          },
+        ],
+      };
+    });
+
+    it('should create a pull request without duplicate reviewer and assignee identities', async () => {
+      // Arange
+      const mockGetUserId = jest.spyOn(client, 'getUserId').mockResolvedValue('my-user-id');
+      const mockResolveIdentityId = jest
+        .spyOn(client, 'resolveIdentityId')
+        .mockImplementation(async (identity?: string) => {
+          return identity || '';
+        });
+      const mockRestApiPost = jest
+        .spyOn(client as any, 'restApiPost')
+        .mockResolvedValueOnce({
+          commits: [{ commitId: 'new-commit-id' }],
+        })
+        .mockResolvedValueOnce({
+          pullRequestId: 1,
+        });
+      const mockRestApiPatch = jest.spyOn(client as any, 'restApiPatch').mockResolvedValueOnce({
+        count: 1,
+      });
+
+      // Act
+      pr.assignees = ['user1', 'user2'];
+      pr.reviewers = ['user1', 'user3'];
+      const pullRequestId = await client.createPullRequest(pr);
+
+      // Assert
+      expect(mockRestApiPost).toHaveBeenCalledTimes(2);
+      expect((mockRestApiPost.mock.calls[1] as any)[1].reviewers.length).toBe(3);
+      expect((mockRestApiPost.mock.calls[1] as any)[1].reviewers).toContainEqual({
+        id: 'user1',
+        isRequired: true,
+        isFlagged: true,
+      });
+      expect((mockRestApiPost.mock.calls[1] as any)[1].reviewers).toContainEqual({
+        id: 'user2',
+        isRequired: true,
+        isFlagged: true,
+      });
+      expect((mockRestApiPost.mock.calls[1] as any)[1].reviewers).toContainEqual({ id: 'user3' });
+      expect(pullRequestId).toBe(1);
+    });
+  });
+});

--- a/extension/tasks/dependabotV2/utils/azure-devops/AzureDevOpsWebApiClient.ts
+++ b/extension/tasks/dependabotV2/utils/azure-devops/AzureDevOpsWebApiClient.ts
@@ -189,7 +189,7 @@ export class AzureDevOpsWebApiClient {
       if (pr.assignees?.length > 0) {
         for (const assignee of pr.assignees) {
           const identityId = isGuid(assignee) ? assignee : await this.resolveIdentityId(assignee);
-          if (identityId) {
+          if (identityId && !allReviewers.some((r) => r.id === identityId)) {
             allReviewers.push({
               id: identityId,
               isRequired: true,
@@ -203,7 +203,7 @@ export class AzureDevOpsWebApiClient {
       if (pr.reviewers?.length > 0) {
         for (const reviewer of pr.reviewers) {
           const identityId = isGuid(reviewer) ? reviewer : await this.resolveIdentityId(reviewer);
-          if (identityId) {
+          if (identityId && !allReviewers.some((r) => r.id === identityId)) {
             allReviewers.push({
               id: identityId,
             });

